pkg/clusterapi/localcontrolplane.go

```diff
@@ -0,0 +1,158 @@
+package clusterapi
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/klog/v2"
+	capav1beta1 "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta1"
+	capav1 "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+	capzv1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
+	clusterv1alpha3 "sigs.k8s.io/cluster-api/api/v1alpha3" //nolint:staticcheck
+	clusterv1alpha4 "sigs.k8s.io/cluster-api/api/v1alpha4" //nolint:staticcheck
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/openshift/installer/cmd/openshift-install/command"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+)
+
+func init() {
+	utilruntime.Must(clusterv1alpha3.AddToScheme(scheme.Scheme))
+	utilruntime.Must(clusterv1alpha4.AddToScheme(scheme.Scheme))
+	utilruntime.Must(clusterv1.AddToScheme(scheme.Scheme))
+	utilruntime.Must(capav1beta1.AddToScheme(scheme.Scheme))
+	utilruntime.Must(capav1.AddToScheme(scheme.Scheme))
+	utilruntime.Must(capzv1.AddToScheme(scheme.Scheme))
+}
+
+// localControlPlane creates a local capi control plane
+// to use as a management cluster.
+// TODO: Add support for existing management cluster.
+type localControlPlane struct {
+	Env            *envtest.Environment
+	Client         client.Client
+	Cfg            *rest.Config
+	BinDir         string
+	KubeconfigPath string
+}
+
+// Name returns the human-friendly name of the asset.
+func (c *localControlPlane) Name() string {
+	return "Local Control Plane"
+}
+
+// Run launches the local control plane.
+func (c *localControlPlane) Run(clusterID *installconfig.ClusterID, installConfig *installconfig.InstallConfig) error {
+	// Create a temporary directory to unpack the cluster-api binaries.
+	binDir, err := os.MkdirTemp("", "openshift-cluster-api-bins")
```

Comment:
We have seen in the past that the /tmp [TempDir](https://pkg.go.dev/os#TempDir) can be mounted with noexec (see https://github.com/openshift/installer/pull/5715 and [CORS-1952](https://issues.redhat.com/browse/CORS-1952)). We should extract these to the install dir.

Reply:
Done!